tools/pythonpkg/tests/fast/types/test_datetime_datetime.py

```diff
@@ -0,0 +1,57 @@
+import duckdb
+import datetime
+import pytest
+
+def create_query(positive, type):
+    inf = 'infinity' if positive else '-infinity'
+    return f"""
+        select '{inf}'::{type}
+    """
+
+class TestDateTimeDateTime(object):
+    @pytest.mark.parametrize('positive', [
+        True,
+        False
+    ])
+    @pytest.mark.parametrize('type', [
+        'TIMESTAMP',
+        'TIMESTAMP_S',
+        'TIMESTAMP_MS',
+        'TIMESTAMP_NS',
+        'TIMESTAMPTZ',
+        'TIMESTAMP_US',
+    ])
+    def test_timestamp_infinity(self, positive, type):
+        con = duckdb.connect()
+
+        expected_val = datetime.datetime.max if positive else datetime.datetime.min
+        query = create_query(positive, type)
+        if positive and type in [
+                'TIMESTAMP_NS'
+        ]:
+            # This query is not supported in core duckdb
+            with pytest.raises(duckdb.ConversionException):
+                con.execute(query).fetchall()
+        elif not positive and type in [
+                'TIMESTAMP_NS',
+                'TIMESTAMP_MS',
+                'TIMESTAMP_S'
+        ]:
+            # This query is not supported in core duckdb but doesn't deterministically produce an error
```

Comment:
Any idea why it doesn't deterministically error?

Reply:
No clue, I think it has to do with an earlier under/overflow from an unchecked expression.
I noticed INFINITY and -INFINITY are not really implemented for TIMESTAMP_S/MS/NS, as I am having to check for the truncated versions of the `timestamp_t::infinity` value.
```c++
struct timestamp_ms_t : public timestamp_t { // NOLINT
	static constexpr timestamp_t infinity() {
		return timestamp_t(int64_t(NumericLimits<int64_t>::Maximum() / 1000) * 1000);
	}
	static constexpr timestamp_t ninfinity() {
		return timestamp_t(int64_t(-NumericLimits<int64_t>::Maximum() / 1000) * 1000);
	}
};
```

So it might be a result of that